app/src/geo/google.js

```diff
@@ -21,43 +65,27 @@ module.exports = {
     },
 
     getAddress: function(location, callback) {
-        geocoder.reverse(location, function(err, geocodeResult) {
-            if (err) log.error(err);
+        var cacheKey = location.lat + "-" + location.lon;
+        addrCache.get(cacheKey, function(err, addr) {
+            if(err) {
+                log.error(err);
+            }
 
-			var res = {};
-			if(geocodeResult && geocodeResult.length > 0) {
-				//we found it
-				res.addr = config.locale.addressformat
-				.replace(/%n/, geocodeResult[0].streetNumber || '')
-				.replace(/%S/, geocodeResult[0].streetName || '')
-				.replace(/%z/, geocodeResult[0].zipcode || '')
-				.replace(/%P/, geocodeResult[0].country || '')
-				.replace(/%p/, geocodeResult[0].countryCode || '')
-				.replace(/%c/, geocodeResult[0].city || '')
-				.replace(/%T/, geocodeResult[0].state || '')
-				.replace(/%t/, geocodeResult[0].stateCode || '');
-				res.streetNumber = geocodeResult[0].streetNumber || '';
-				res.streetName = geocodeResult[0].streetName || '';
-				res.zipcode = geocodeResult[0].zipcode || '';
-				res.country = geocodeResult[0].country || '';
-				res.countryCode = geocodeResult[0].countryCode || '';
-				res.city = geocodeResult[0].city || '';
-				res.state = geocodeResult[0].state || '';
-				res.stateCode = geocodeResult[0].stateCode || '';
-			} else {
-				//didn't find anything, default it
-				res.addr = '';
-				res.streetNumber = '';
-				res.streetName = '';
-				res.zipcode = '';
-				res.country = '';
-				res.countryCode = '';
-				res.city = '';
-				res.state = '';
-				res.stateCode = '';
-			}
+            if(addr) {
+                parseAddr(err, addr, callback);
+                return;
+            }
 
-            callback(err, res);
+            geocoder.reverse(location, function(gerr, geocodeResult) {
+                if (gerr) log.error(gerr);
```

Comment:
why gerr 🤔

Reply:
It feels dirty outscoping err from the outer promises. Is that a bad thing?